### PR TITLE
Initial authentication setup

### PR DESCRIPTION
**Pull request by legendofa**
Authentication using HTTP Headers.

- [x] Requirements of the issue are met
- [x] Code has been reviewed
- [x] Code is documented
- [x] GraphQL related artefacts are documented

### CODE REVIEW

**Comment by legendofa** [comment]
The `ShoppingCartItem` mutations are not authenticated yet, as it is difficult to access the corresponding user there.

**Comment by delvh** [comment]
> Requirements of the issue are met

Which issue?

**Comment by legendofa** [comment]
> > Requirements of the issue are met
> 
> Which issue?

https://frontend.gropius.duckdns.org/components/6fce3fcc-4ebd-4c13-b396-64501324918d/issues/c86a1fb4-8773-47b0-9895-2d4c976a8d96

**Comment by legendofa** [comment]
> The `ShoppingCartItem` mutations are not authenticated yet, as it is difficult to access the corresponding user there.

Should be solved now

File: src/query.rs [hunk 1]
```diff
@@ -43,7 +46,9 @@ impl Query {
     ) -> Result<ShoppingCartItem> {
         let db_client = ctx.data_unchecked::<Database>();
         let collection: Collection<User> = db_client.collection::<User>("users");
-        query_shoppingcart_item(&collection, id).await
+        let user = query_shoppingcart_item_user(&collection, id).await?;
+        authenticate_user(&ctx, user._id)?;
```
Comment (nk-coding) [inline src/query.rs]: why authentication logic in the entity resolver?

Comment (legendofa) [reply src/query.rs]: Removed now

**Review by nk-coding — APPROVED** [review]

**Pull request merged**